### PR TITLE
feat(glyphs): recently used glyphs functionality AP-341

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖

AP-341
- added state for recently used glyphs including a counter
- removed global style that was hiding typographyText on xs devices
- limit recently used glyph array size on mobile devices
- introduce mq xs breakpoint

AP-318
- add padding to bottom of enhancers glyph list to show all

AP-264
- add small img to reply rather than string of glyph url

**Which issue(s) this PR fixes** 🔨
AP-314, AP-318, AP-264

**Special notes for reviewers** 🗒️
- I'm not a good copywriter, nor do I claim to be. If anyone has something better than 'Try using some glyphs', go for it. This phrase will be displayed when recently used glyphs state is empty

**Additional comments** 🎤
- in regards to UX, do we want to close the enhancers menu after sending a glyph?

### CODE REVIEW

**Review by Jekrimo — APPROVED** [review]

**Comment by stavares843** [comment]
AP-314, AP-318, AP-264 - fixed ✅

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
do you want to add this one as well? 😂  [AP-359](https://satellite-im.atlassian.net/browse/AP-359)

File: components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue [hunk 1]
```diff
@@ -92,7 +92,7 @@ export default Vue.extend({
               requestWidth = '300'
             }
             const facebookOutSource = `https://www.facebook.com/plugins/video.php?href=${encodeURIComponent(
-              link.split('?')[0]
+              link.split('?')[0],
```
Comment (WanderingHogan) [inline components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue:95]: weird place to put a comma

File: components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue [hunk 2]
```diff
@@ -34,7 +34,7 @@ export default Vue.extend({
     parseEmbeddableVideoContentFromText(messagetext: string) {
       // parse incoming text for links
       const arrayOfLinks = messagetext.match(
-        /(\b(https?):\/\/[-A-Z0-9+&@#/%?=~_|!:,.;]*[-A-Z0-9+&@#/%=~_|])/gi
+        /(\b(https?):\/\/[-A-Z0-9+&@#/%?=~_|!:,.;]*[-A-Z0-9+&@#/%=~_|])/gi,
```
Comment (WanderingHogan) [inline components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue:37]: can you move this to the config

File: components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue [hunk 3]
```diff
@@ -120,15 +120,15 @@ export default Vue.extend({
...
+                `https://player.twitch.tv/?${videoId}&parent=${window.location.hostname}&autoplay=false`,
               ),
               aspectRatioClass: 'iframe-container-16-9',
             }
           }
 
           if (
             link.match(
-              /^https?:\/\/([a-z0-9-]+[.])spotify[.]com\/(playlist|embed)?/g
+              /^https?:\/\/([a-z0-9-]+[.])spotify[.]com\/(playlist|embed)?/g,
```
Comment (WanderingHogan) [inline components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue:131]: add to config so regex are all in one place

File: store/ui/types.ts [hunk 4]
```diff
@@ -29,9 +30,15 @@ export interface EnhancerInfo {
 }
 
 export interface EmojiUsage {
-  code: string,
-  count: number,
-  content: string,
+  code: String
```
Comment (WanderingHogan) [inline store/ui/types.ts:33]: use lower case primitives

**Review by WanderingHogan — CHANGES_REQUESTED** [review]
Good, couple minor changes though when you get a chance next week

File: components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue (hunk 1 above)
Comment (josephmcg) [reply components/ui/Chat/Embeds/EmbeddedLinkContent/EmbeddedLinkContent.vue:95]: yeah, bad prettier

**Comment by josephmcg** [comment]
> AP-359

lol I'll do that one separately since this is already tackling a bunch of issues. I think it was introduced with the swiper stuff

**Comment by stavares843** [comment]
neat! 🎉

**Comment by stavares843** [comment]
just added the missing dev review label because of @WanderingHogan review

**Review by WanderingHogan — APPROVED** [review]

**Pull request merged**